internal/ast/utilities.go

```diff
@@ -885,7 +885,8 @@ func newParentInChildrenSetter() func(node *Node) bool {
 	}
 
 	state.visit = func(node *Node) bool {
-		if state.parent != nil {
+		if state.parent != nil && node.Parent != state.parent {
```

Comment:
Does this actually avoid a race? A write can happen somewhere other than this, and then it's a read-write race.

Reply:
The only other concurrent write is what I protected in #970

Reply:
Right, but then this part of the code doesn't need to change, since if there were a race, it'd be a race either way, right? We're just incurring an extra read before the write